python/src/dataset/io_stats.rs

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+//! IO statistics tracking for dataset operations
+
+use pyo3::{pyclass, pymethods};
+
+/// IO statistics for dataset operations
+///
+/// This tracks the number of IO operations and bytes transferred for read and write
+/// operations performed on the dataset's object store.
+///
+/// Note: Calling `io_stats()` returns the statistics accumulated since the last call
+/// and resets the internal counters (incremental stats pattern).
+#[pyclass(name = "IOStats", module = "_lib", get_all)]
+#[derive(Clone, Debug)]
+pub struct IoStats {
+    /// Number of read IO operations performed
+    pub read_iops: u64,
+    /// Total bytes read from storage
+    pub read_bytes: u64,
+    /// Number of write IO operations performed
+    pub write_iops: u64,
+    /// Total bytes written to storage
+    pub write_bytes: u64,
+    /// Number of disjoint periods where at least one IO is in-flight
+    ///
+    /// This metric helps understand IO parallelism. A lower number indicates
+    /// more parallel IO operations.
+    pub num_hops: u64,
```

Comment:
calling these hops/network hops isn't clear to me. Would this be network hops behind the S3 endpoint? That's how I would interpret it but I don't think we have that info